packages/backend/src/tools/updateDiffHistory.ts

```diff
@@ -36,14 +37,42 @@ async function updateDiffHistoryFile() {
     getFileVersionOnMainBranch(`${discoveryFolder}/discovered.json`)
   const discoveryFromMainBranch =
     DiscoveryJsonFromMainBranch === ''
-      ? { contracts: [] }
+      ? undefined
       : (JSON.parse(DiscoveryJsonFromMainBranch) as DiscoveryOutput)
   const diff = diffDiscovery(
-    discoveryFromMainBranch.contracts,
+    discoveryFromMainBranch?.contracts ?? [],
     curDiscovery.contracts,
     config,
   )
 
+  // To check for changes to source code,
+  // download sources for block number from main branch
+  let codeDiff = ''
+  if (discoveryFromMainBranch) {
+    const root = `discovery/${projectName}/${chainName}`
+    // Remove any old sources we fetched before, so that their count doesn't grow
+    await rimraf(`${root}/.code@*`, { glob: true })
+
+    const blockNumberFromMainBranch = discoveryFromMainBranch.blockNumber
+    const cli = [
```

Comment:
This feels bad but robust at the same time. WDYT about exposing and using the source code fetching API from discovery?